src/m365/spo/commands/page/page-copy.ts

```diff
@@ -0,0 +1,163 @@
+import * as chalk from 'chalk';
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import { ClientSidePageProperties } from './ClientSidePageProperties';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  sourceName: string;
+  targetUrl: string;
+  webUrl: string;
+  overwrite?: boolean;
+}
+
+class SpoPageCopyCommand extends SpoCommand {
+  public get name(): string {
+    return `${commands.PAGE_COPY}`;
+  }
+
+  public get description(): string {
+    return 'Creates a copy of a modern page or template';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Id', 'PageLayoutType', 'Title', 'Url'];
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.sourceName = typeof args.options.sourceName !== 'undefined';
```

Comment:
Since `sourceName` and `targetUrl` are required, we don't need to track their usage in the telemetry because they would always show up as set.